apps/system/js/child_window_factory.js

```diff
@@ -157,10 +157,16 @@
     if (!this.app.isActive() || this.app.isTransitioning()) {
       return false;
     }
+
+    var iframe = evt.detail.frameElement;
+
+    // This new window should be allowed to go full screen.
+    iframe.setAttribute('mozallowfullscreen', 'true');
```

Comment:
Shouldn't we get this value from the parent window, instead of setting it always to true?

```javascript
var allowFullscreen = evt.target.getAttribute('mozallowfullscreen');
iframe.setAttribute('mozallowfullscreen', allowFullscreen);
```